File: server/src/routes/requirement.routes.ts
import { Router } from 'express';
import { authMiddleware } from '../middlewares/auth.middleware.js';
import {
  createRequirementBaselineController,
  createRequirementController,
  deleteRequirementController,
  extractRequirementsController,
  getRequirementController,
  listProjectRequirementsController,
  listRequirementVersionsController,
  updateRequirementController,
} from '../controllers/requirement.controller.js';

export const requirementRouter = Router();

requirementRouter.use(authMiddleware);
requirementRouter.post('/extract', extractRequirementsController);
requirementRouter.post('/baseline', createRequirementBaselineController);
requirementRouter.get('/versions/:projectId', listRequirementVersionsController);
requirementRouter.get('/project/:projectId', listProjectRequirementsController);
requirementRouter.post('/', createRequirementController);
requirementRouter.get('/:requirementId', getRequirementController);
requirementRouter.patch('/:requirementId', updateRequirementController);
requirementRouter.delete('/:requirementId', deleteRequirementController);

File: server-go/internal/modules/requirement/requirement_routes.go
package requirement

import (
	"driftledger/server-go/internal/config"
	"driftledger/server-go/internal/middleware"
	"github.com/gin-gonic/gin"
	"go.mongodb.org/mongo-driver/mongo"
)

func RegisterRoutes(group *gin.RouterGroup, db *mongo.Database, cfg config.Config) {
	handler := NewHandler(NewService(db))
	group.Use(middleware.Auth(db, cfg))
	group.GET("/project/:projectId", handler.ListByProject)
	group.POST("", handler.Create)
	group.POST("/extract", handler.Extract)
	group.POST("/baseline", handler.Baseline)
	group.GET("/versions/:projectId", handler.Versions)
	group.GET("/:requirementId", handler.Get)
	group.PATCH("/:requirementId", handler.Update)
	group.DELETE("/:requirementId", handler.Delete)
}
